shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -191,9 +192,11 @@ public void surfaceDestroyed(SurfaceHolder holder) {
         PlatformPlugin platformPlugin = new PlatformPlugin(activity, platformChannel);
         addActivityLifecycleListener(platformPlugin);
         mImm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
-        mTextInputPlugin = new TextInputPlugin(this, dartExecutor);
+        PlatformViewsResolver platformViewsResolver = mNativeView.getPluginRegistry().getPlatformViewsController();
+        mTextInputPlugin = new TextInputPlugin(this, dartExecutor, platformViewsResolver);
         androidKeyProcessor = new AndroidKeyProcessor(keyEventChannel, mTextInputPlugin);
         androidTouchProcessor = new AndroidTouchProcessor(flutterRenderer);
+        mNativeView.getPluginRegistry().getPlatformViewsController().attachTextInputPlugin(mTextInputPlugin);
```

Comment:
I'm a bit concerned about this relationship between `PlatformViewsController` and `TextInputPlugin`. While `TextInputPlugin` technically requires a `PlatformViewsResolver` in its constructor, we know that really its the `PlatformViewsController` that we're passing. So effectively `TextInputPlugin` requires a `PlatformViewsController` at construction time, and then the `PlatformViewsController` requires someone to instruct it to attach to the `TextInputPlugin`. 

Why not fold the relationship between `TextInputPlugin` and `PlatformViewsResolver` into the required relationship between `TextInputPlugin` and `PlatformViewsController`, i.e., just use the attached `PlatformViewsController` as the `PlatformViewsResolver`? Are there any cases where a `PlatformViewsResolver` would be required by the `TextInputPlugin` when that `TextInputPlugin` is not attached to a `PlatformViewsController`?

Reply:
Deleted PlatformViewsResolver, and passed a PlatformViewsController to TextInputPlugin.
The TextInputPlugin attaches itself to the PlatformViewsController.

I'm expecting that in the new embedding the TextInputPlugin will be detached from the PlatformViewsController in `detachFromFlutterEngine`(I guess when we do that we should probably add a `dispose` method or something like that to TextInputPlugin).